gradle-plugin/src/main/java/com/squareup/anvil/plugin/AnvilSubplugin.kt

```diff
@@ -42,40 +40,13 @@ class AnvilSubplugin : KotlinGradleSubplugin<AbstractCompile> {
     // for this specific compile task will be included in the task output. The output of different
     // compile tasks shouldn't be mixed.
     val srcGenDir = File(project.buildDir, "anvil${File.separator}src-gen-${kotlinCompile.name}")
-    val srcGenDirPath = srcGenDir.absolutePath
-    val srcGenDirOption = SubpluginOption(
-        key = "src-gen-dir",
-        value = srcGenDirPath
-    )
-
-    project.afterEvaluate {
-      // Notice that we pass the absolutePath to the Kotlin compiler plugin. That is necessary,
-      // because the plugin has no understanding of Gradle or what the working directory is. The
-      // Kotlin Gradle plugin adds all SubpluginOptions as input to the Gradle task. This is bad,
-      // because now the hash of inputs is different on every machine. We override this input with
-      // a relative path in order to preserve some safety and avoid the cache misses.
-      val key = "${getCompilerPluginId()}.${srcGenDirOption.key}"
-      val relativePath = srcGenDir.relativeTo(project.buildDir).path
-
-      project.tasks.withType(KotlinCompile::class.java).configureEach { task ->
-          task.inputs.property(key, relativePath)
-      }
-      project.tasks.withType(KaptTask::class.java).configureEach { task ->
-          // We don't apply the compiler plugin for KaptTasks. But for some reason the Kotlin Gradle
-          // plugin copies the inputs with the "kotlinCompile" prefix.
-          task.inputs.property("kotlinCompile.$key", relativePath)
-      }
-      project.tasks.withType(KaptGenerateStubsTask::class.java).configureEach { task ->
-          // We don't apply the compiler plugin for KaptTasks. But for some reason the Kotlin Gradle
-          // plugin copies the inputs with the "kotlinCompile" prefix.
-          task.inputs.property("kotlinCompile.$key", relativePath)
-      }
-    }
```

Comment:
The stupid workaround is gone.

Reply:
Just out of curiosity, why is it no longer necessary?

Reply:
Because the file path isn't added to the task input anymore in the first place. Instead of using `SubpluginOption` I'm using now `FilesSubpluginOption`, which values aren't added.